### PR TITLE
refactor(ENTESB-12218): Minimize 3.x specifications for generated API connector

### PR DESCRIPTION
**Pull request by christophd**
Generated rest-swagger connector just uses a subset of the OpenAPI specification. Optimizer now also minimizes 3.x specs removing all unused information.

### CODE REVIEW

File: app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java [hunk 1]
```diff
@@ -61,6 +61,7 @@
     protected final String specification;
 
     public BaseOpenApiGeneratorExampleTest(final String connectorQualifier, final String name, final String version) throws IOException {
+        System.out.println(name);
```
Comment (zregvart) [inline app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java]:
```suggestion
```

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/util/SpecificationOptimizer.java [hunk 2]
```diff
@@ -47,7 +47,12 @@ private SpecificationOptimizer() {
      */
     public static String minimizeForComponent(final OasDocument openApiDoc) {
         final ObjectNode json = (ObjectNode) Library.writeNode(openApiDoc);
-        json.remove(Arrays.asList("info", "tags", "definitions", "externalDocs"));
+        json.remove(Arrays.asList("info", "tags", "definitions", "responses", "externalDocs"));
+
+        final JsonNode components = json.get("components");
+        if (components != null) {
+            ((ObjectNode)components).remove(Arrays.asList("schemas", "responses", "requestBodies", "examples", "headers", "links", "callbacks"));
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/util/SpecificationOptimizer.java]: Just to make sure, `headers` is not the global parameters from OpenAPI 2, i.e. it's not shared across operations. The global parameters in OpenAPI 2 were used by the Camel component to include them in the request (for any operation).

File: app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java [hunk 3]
```diff
@@ -81,6 +82,8 @@ public void shouldGenerateAsExpected() throws IOException {
         final Map<String, String> expectedConfiguredProperties = expected.getConfiguredProperties();
         final String expectedSpecification = expectedConfiguredProperties.get("specification");
 
+        System.out.println(generatedSpecification);
```
Comment (zregvart) [inline app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java]:
```suggestion
```

File: app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java (hunk 1 above)
Comment (christophd) [reply app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java]: Oh did that for debugging reasons. Thanks!

File: app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java (hunk 3 above)
Comment (christophd) [reply app/server/api-generator/src/test/java/io/syndesis/server/api/generator/openapi/BaseOpenApiGeneratorExampleTest.java]: Left over from debugging sessions. Thanks!

**Pull request merged**